Add Alias `@pmf` for `@pmFromFile`

satwiksps opened the pull request:
# Add Alias `@pmf` for `@pmFromFile`

## Description
This pull request introduces the short alias `@pmf` for the operator `@pmFromFile` in accordance with the documentation. The alias provides a shorthand for users while maintaining the same functionality as `@pmFromFile`.

### Changes Made
1. **Alias Registration:**
   - Added `@pmf` alias to the `pmFromFile` operator by updating the `Register()` call in the `pm_from_file.go` file.
   
2. **Test Suite:**
   - Introduced a new test case in `pm_from_file_test.go` to validate both `@pmFromFile` and its alias `@pmf`. 
   - Tests include:
     - Valid inputs
     - Invalid inputs
     - Missing file handling
   - Mocked the `loadFromFile` function for unit testing without requiring actual files.

### Motivation
The `@pmf` alias was described in the [documentation](https://coraza.io/docs/seclang/operators/#pmf) but was missing in the implementation. This PR aligns the implementation with the documentation and improves usability by providing a shorthand for `@pmFromFile`.

### Testing
- Verified that both `@pmFromFile` and `@pmf` operators work as expected:
  - Properly match patterns loaded from files.
  - Return expected results for valid and invalid inputs.
  - Handle missing or invalid files gracefully.
- Added unit tests using `go test` to ensure robust validation of functionality.

### Related Issues
- Resolves [#1283](https://github.com/corazawaf/coraza/issues/1283).

### Checklist
- [x] Updated the test suite with cases for `@pmf`.
- [x] Implemented the `@pmf` alias in the code.
- [x] Ran all tests to verify correctness.

---

jptosso commented:
Thank you for your contribution. I don't think we need new tests for pmfromfile, we should just test that the pmFromFile pointer is the same as pm

satwiksps commented:
> Thank you for your contribution. I don't think we need new tests for pmfromfile, we should just test that the pmFromFile pointer is the same as pm

# Add Alias for pmFromFile and Validate Equivalence with pm
- Implemented the @pmf alias for pmFromFile as described in the documentation.
- Added a test to confirm that pmFromFile behaves identically to pm (shared behavior and logic).
- Removed redundant tests for pmFromFile to avoid duplication.

I am new to Open Source Contributions.
Please guide me if I need to make any additional changes.

M4tteoP commented:
Hi, thanks for the PR! Some CI tests are failing:
- the linter: for this one you probably have to format the new file created. To do so you can run `mage format` from the top directory of the project. Then, you can locally check with `mage lint` to see if everything passes 
```
internal/operators/pm_from_file.go:54: File is not `gofmt`-ed with `-s` (gofmt)
    Register("pmFromFile", newPMFromFile)
    Register("pmf", newPMFromFile)
internal/operators/pm_from_file_test.go:1: File is not `gofmt`-ed with `-s` (gofmt)
package operators
internal/operators/pm_from_file_test.go:[5](https://github.com/corazawaf/coraza/actions/runs/12913855172/job/36012093907?pr=1291#step:4:6): File is not `goimports`-ed (goimports)
    "github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
exit status 1
Error: running "go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.[6](https://github.com/corazawaf/coraza/actions/runs/12913855172/job/36012093907?pr=1291#step:4:7)0.3 run" failed with exit code 1
exit status 1
```
- The added test:
```
--- FAIL: TestPmFromFileAlias (0.00s)
    pm_from_file_test.go:20: Failed to initialize pmFromFile: empty dirs
FAIL
```

fzipi commented on internal/operators/pm_from_file_test.go:
```diff
@@ -0,0 +1,27 @@
+
+    "github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+)
+func TestPmFromFileAlias(t *testing.T) {
```
Does this describe the actual test?

I think the test only compares the `pm` and `pmfromfile` operators. Not the alias.

M4tteoP replied:
@satwiksps This test can easily be extended testing the alias alongside `@pmfromfile` https://github.com/corazawaf/coraza/blob/9293be843709277bf32ca6e2cada40775a19ab4a/testing/engine/operators_with_files.go

fzipi reviewed (changes requested):
Probably we want to add a test at the secrule level to see that is gets recognized properly?

satwiksps commented:
Thank you for reviewing my PR. I am working on the changes.
